src/utils/scores.test.ts

```diff
@@ -120,25 +120,45 @@ describe('score tests', () => {
     ORDER BY num_comments DESC
     ;`;
 
-    const discussionCommentersQuery = `
+    const githubCommentersQuery = `
```

Comment:
Way out of scope for this PR, but I think we should consider exploring in-memory SQLite databases for testing, rather than matching these super long queries. Reasoning:

1. Most of the work this code is testing is actually in the query. So basically, this test boils down to "is the query spelled how it's spelled?". But we don't want to test the spelling of the query - we want to test the underlying semantics (given a table with these N rows, did it pick the subset S that we expect?).
2. My understanding is that BiqQuery SQL is a superset of SQLite's implementation of the SQL language. This is good: we can rely only on a small set of well-implemented and standardized features, rather than locking ourselves into the BigQuery ecosystem. BigQuery's extensions are probably very useful for efficient large scale (millions of rows) data processing, but at the scale we're operating at, using slightly less efficient SQL for greater portability and testability seems like a good trade.
3. I think tests written using an in-memory fake will be easier to read and verify for correctness. Reading through this query right now, I *think* it looks correct, but I'm basically just reading it verbatim and guessing at what it will do. If instead the test script was something like "create new in-memory DB, add 4 rows, run test, ensure only 2 rows showed up", it would be a lot easier to offer feedback ("you should add this kind of row", "it seems like your missing this kind of data", "we should add a row at the boundary of the `<= 7` condition", etc).

Anyway, what I propose is a rather large refactor, so I'm not sure it needs to be done now (or maybe ever), but just something to think about. Maybe a useful pattern to try for some simple query, either here or in the future, just to see how it feels?

Reply:
Yeah, you have brought this up before. I do think it would be nice to have but something that I don't think is worth it right now